posydon/binary_evol/DT/step_detached.py

```diff
@@ -819,14 +818,14 @@ def get_MESA_labels(list_for_matching):
             # if matching is still not successful, set result to NaN:
             if (np.abs(sol.fun) > tolerance_matching_integration_hard or not sol.success):
                 if self.verbose:
-                    print("\nMatching result is NOT successful, with ",
-                          np.abs(sol.fun), ">", tolerance_matching_integration, "tolerance")
+                    print("\nMatching result is NOT successful, with tolerance ",
+                          np.abs(sol.fun), ">", tolerance_matching_integration_hard)
```

Comment:
Yes, so I did try rewriting this so that the "hard" limit is tested against the _best_ matching solution of all the attempts. However, this was creating some new issues with the matching results, so I decided that reworking this should be done in a different PR.

I asked Manos about this "hard" limit and all he said was that it was meant to apply to the stripped_He stars to allow less of them to fail in the matching, so maybe going forward that is something we should try to implement correctly (in addition to reworking the matching for these stars in general as we discussed).

Reply:
Maybe I wasn't clear enough, what I'd suggest as solution:
When can check against the stricter limit for each try. But instead of always replacing the solution by a subsequent try, I'd suggest to keep the better solution of the previous and current try. In this way we would compare the lest strict limit not against the last try only, but against the best try.
To do that, I'd not overwrite the data from the previous try in a new one, but instead create new variables in the new try and overwrite the previous try only if the new try has a better solution. So something like
```
sol = get_sol()
if sol.fun>tolerance:
    new_sol = get_sol2()
    if sol.fun>new_sol.fun:
        sol = new_sol
if sol.fun>tolerance:
    new_sol = get_sol3()
    if sol.fun>new_sol.fun:
        sol = new_sol
```

Reply:
Yes, that is what I did!

Reply:
Ok I've created an issue for this: #447